integration-cli/docker_cli_pull_test.go

```diff
@@ -757,3 +758,34 @@ func (s *DockerRegistriesSuite) TestPullFromBlockedRegistry(c *check.C) {
 		c.Fatalf("expected %s and %s to have the same size (%s != %s)", bb2Img.name, bbImg.name, bb2Img.size, bbImg.size)
 	}
 }
+
+func (s *DockerRegistriesSuite) TestPullNeedsAuth(c *check.C) {
+	c.Assert(s.d.StartWithBusybox("--add-registry="+s.regWithAuth.url), check.IsNil)
+
+	repo := fmt.Sprintf("%s/runcom/busybox", s.regWithAuth.url)
+	repoUnqualified := "runcom/busybox"
+
+	out, err := s.d.Cmd("tag", "busybox", repo)
+	c.Assert(err, check.IsNil, check.Commentf(out))
+
+	// this means it needs auth...
+	resp, err := http.Get(fmt.Sprintf("http://%s/v2/", s.regWithAuth.url))
+	c.Assert(err, check.IsNil)
+	c.Assert(resp.StatusCode, check.Equals, http.StatusUnauthorized)
+
+	// login with the registry...
+	out, err = s.d.Cmd("login", "-u", "testuser", "-p", "testpassword", "-e", "test@test.org", s.regWithAuth.url)
+	c.Assert(err, check.IsNil, check.Commentf(out))
+
+	// push the image so we can pull from unqualified "runcom/busybox"
+	out, err = s.d.Cmd("push", repo)
+	c.Assert(err, check.IsNil, check.Commentf(out))
+
+	out, err = s.d.Cmd("pull", repoUnqualified)
+	c.Assert(err, check.IsNil, check.Commentf(out))
+	// check it's pulling from private reg with auth with unqualified images
+	expected := fmt.Sprintf("Trying to pull repository %s ... latest: Pulling from %s", repo, repoUnqualified)
+	if !strings.Contains(out, expected) {
```

Comment:
Please confirm that no attempt to pull from `docker.io` is done. Or that resulting image is prefixed with expected repo name.